test/lsp/cache_protocol_test_corpus.cc

```diff
@@ -605,4 +605,70 @@ TEST_CASE_FIXTURE(CacheProtocolTest, "ReopenExistingSessionCacheDir") {
     REQUIRE(!FileOps::exists(sessionPath));
 }
 
+TEST_CASE_FIXTURE(CacheProtocolTest, "ReapOldCacheDirectories") {
+    auto opts = lspWrapper->opts;
+    REQUIRE(!opts->cacheDir.empty());
+
+    // We don't need any additional LSP behavior here
+    lspWrapper = nullptr;
+
+    // We should start in a state where there are no cache directories.
+    REQUIRE(FileOps::listSubdirs(opts->cacheDir).empty());
+
+    // Create a few fake cache directories to simulate crashes. We test for existing pids to avoid faking a directory
+    // for an unrelated pid.
+    vector<string> caches;
+    {
+        auto needed = 5;
+        auto pid = 120;
+        while (needed > 0) {
+            ++pid;
+
+            if (kill(pid, 0) == 0) {
+                continue;
+            } else if (errno != ESRCH) {
+                continue;
+            }
+
+            --needed;
```

Comment:
This loop seems like it could take a very long time to create the needed files; looking at Stripe devboxes, they have a lot of PIDs in this range.

I don't know what a better answer would be -- we could start at the test's `getpid()`, maybe?